test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java

```diff
@@ -0,0 +1,421 @@
+package org.corfudb.runtime.checkpoint;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.reflect.TypeToken;
+import org.corfudb.protocols.logprotocol.CheckpointEntry;
+import org.corfudb.protocols.logprotocol.SMREntry;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+import org.corfudb.runtime.CheckpointWriter;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.LogUnitClient;
+import org.corfudb.runtime.collections.SMRMap;
+import org.corfudb.runtime.object.transactions.TransactionType;
+import org.corfudb.runtime.view.AbstractViewTest;
+import org.corfudb.runtime.view.stream.BackpointerStreamView;
+import org.corfudb.util.serializer.Serializers;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Basic smoke tests for checkpoint-in-stream PoC.
+ */
+
+public class CheckpointSmokeTest extends AbstractViewTest {
+
+    public CorfuRuntime r;
+
+    @Before
+    public void setRuntime() throws Exception {
+        // This module *really* needs separate & independent runtimes.
+        r = getDefaultRuntime().connect(); // side-effect of using AbstractViewTest::getRouterFunction
+        r = new CorfuRuntime(getDefaultEndpoint()).connect();
+    }
+
+    /** First smoke test, steps:
+     *
+     * 1. Put a couple of keys into an SMRMap "m"
+     * 2. Write a checkpoint (3 records total) into "m"'s stream.
+     *    The SMREntry records in the checkpoint will *not* match
+     *    the keys written by step #1.
+     * 3. Put a 3rd key into "m".
+     * 4. The current runtime should see all original values of
+     *    the three put() keys.
+     * 5. Open a new runtime and a new map "m2" on the same stream.
+     *    Verify via get() that we cannot see the first two keys,
+     *    we see the snapshot's keys 7 & 8, and we can also see
+     *    key 3.
+     *
+     * This is not correct map behavior: keys shouldn't be lost like
```

Comment:
If checkpoints work correctly, they're invisible at nearly every layer of the system ... and that's a problem for testing.  I want to make it very clear to the reader that this test is doing something weird, and it's intentional.  This test case won't evolve over time, it should stay weird.